pkg/resourceskeleton/testdata/uri-skeleton.yaml

```diff
@@ -1111,34 +1111,30 @@
     location: us-central1
   ResourceConfigId: EdgeContainerVpnConnection
   URI: "https://edgecontainer.googleapis.com/v1/projects/kcc-test/locations/us-central1/vpnConnections/edgecontainervpnconnection-test"
-- ExpectedSkeleton:
-    apiVersion: vertexai.cnrm.cloud.google.com/v1beta1
-    kind: VertexAIDataset
-    spec:
-      projectRef:
-    external: kcc-test
-    location: us-central1
+- ExpectedSkeleton: null # VertexAIDataset does not support import
   ResourceConfigId: VertexAIDataset
-  URI: "https://us-central1-aiplatform.googleapis.com/v1beta1/projects/kcc-test/locations/us-central1/datasets/1234567890?alt=json"
+  URI: ""
 - ExpectedSkeleton:
     apiVersion: vertexai.cnrm.cloud.google.com/v1beta1
     kind: VertexAIEndpoint
     metadata:
       name: vertexaiendpoint-test
     spec:
       projectRef:
-    external: kcc-test
-    region: us-central1
+        external: kcc-test
+      region: us-central1
   ResourceConfigId: VertexAIEndpoint
   URI: "https://us-central1-aiplatform.googleapis.com/v1beta1/projects/kcc-test/locations/us-central1/endpoints/vertexaiendpoint-test"
 - ExpectedSkeleton:
     apiVersion: vertexai.cnrm.cloud.google.com/v1beta1
     kind: VertexAIIndex
-    metadata:
-      name: vertexaiindex-test
+    metadata: {}
```

Comment:
Why is the metadata section cleared up?

Reply:
As the result of enabling the test `TestNewFromURI` for resource (by setting `idTemplateCanBeUsedToMatchResourceName` to `true`), this is the expected output of the test. It is an output only field in [API](https://cloud.google.com/vertex-ai/docs/reference/rest/v1beta1/projects.locations.endpoints). So it ended up showing up under `status.observedState.name`. Not sure if I miss configured the service mapping?

Reply:
Thanks! I set `idTemplateCanBeUsedToMatchResourceName: false` for VertexAIIndex.